src/component/2d/zones/Signal.tsx

```diff
@@ -59,7 +59,10 @@ function Signal({ signal, isVisible }: SignalProps) {
     <g className="zone-signal">
       {isVisible.signals && (
         <g>
-          <SignalCrosshair signal={signal} />
+          <SignalCrosshair
+            signal={signal}
+            key={`${signal.x.delta},${signal.y.delta}`}
```

Comment:
React key should not be used for tests, because it has real effects on React rendering

Reply:
Would you please suggest a way to get delta values in test

Also I have a question , can regex be used with CSS locators as well?
In the documentation is used only on react component, and it didn't work in the other cases

Reply:
The values you are interested in are in the `signal` prop. Can't you match using the nested property syntax?
The example from the docs should help you: https://playwright.dev/docs/other-locators#react-locator

Reply:
I seems you can use regex in CSS selectors. They mention it in https://playwright.dev/docs/other-locators#css-matching-by-text

Reply:
> Can't you match using the nested property syntax?

I couldn't do that because in the signal prop we have the exact value of the delta but in the panel we have the rounded value

Reply:
but in the key you don't put the rounded value here

Reply:
> I seems you can use regex in CSS selectors.

Regex here is used the textual content, I used it to match props [https://playwright.dev/docs/other-locators#react-locator](https://playwright.dev/docs/other-locators#react-locator)

Reply:
> I couldn't do that because in the signal prop we have the exact value of the delta but in the panel we have the rounded value

I used the exact value to test, and it's working now thank you

Reply:
I think you need to discuss with @lpatiny, because I don't understand what is important to be tested in this case.

Reply:
What happens when you try a regex with the numerical value?

Reply:
> What happens when you try a regex with the numerical value?

I didn't get any result

I even tried 
```tsx
await expect(
      nmrium.page.locator(
        '_react=Signal[signal.x.delta=/\\.*/][signal.y.delta=/\\.*/]',
      ),
    ).toHaveCount(1);
```
and I got 
```tsx
Expected: 1
Received: 0
```